packages/xpetra/src/CrsMatrix/Xpetra_EpetraCrsMatrix.hpp

```diff
@@ -1234,6 +1238,14 @@ class EpetraCrsMatrixT <int, EpetraNode>
                                "Xpetra::EpetraCrsMatrix::setAllValues is not implemented");
   }
 
+  void residual(const MultiVector< Scalar, LocalOrdinal, GlobalOrdinal, Node > & X,
+                const MultiVector< Scalar, LocalOrdinal, GlobalOrdinal, Node > & B,
+                MultiVector< Scalar, LocalOrdinal, GlobalOrdinal, Node > & R) const {
+    Scalar one = Teuchos::ScalarTraits<Scalar>::one(), negone = -one;
+    apply(X,R);
+    R.update(one,X,negone);
```

Comment:
This will be fixed in subsequent commits.

Reply:
Done